include/taco/tensor.h

```diff
@@ -916,12 +924,32 @@ struct TensorBase::Content {
   Grid partition;
   std::vector<TensorDistribution> distribution;
 
+  // DISTAL runtime stuff.
+  DISTALRuntime::LegionTensor legionTensor;
```

Comment:
It's weird to have this mirroring the `taco::Tensor`. Another API design might either just use `TensorVar`s, or stop allocating a `storage` for normal tensors.